### PR TITLE
Cleanup event-config.h

### LINKED ISSUE #1332: EVENT__HAVE_TAILQFOREACH occurs twice in event-config.h.cmake
https://github.com/libevent/libevent/blob/b19af675c7601a7867f26c33072cda7ea125adb2/event-config.h.cmake#L91
https://github.com/libevent/libevent/blob/b19af675c7601a7867f26c33072cda7ea125adb2/event-config.h.cmake#L410

### PR DESCRIPTION
**Pull request by Coeur**
Fix #1332.
The duplication was introduced with e415196a7da61ba32decf508ae4076d7ae5faf20

Also addresses partially #1329, but I only removed the stuff for which I was 100% sure it was unused.

### CODE REVIEW

File: event-config.h.cmake [hunk 1]
```diff
@@ -427,10 +390,6 @@
 /* Define if timerclear is defined in <sys/time.h> */
 #cmakedefine EVENT__HAVE_TIMERCLEAR 1
 
-/* Define if timercmp is defined in <sys/time.h> */
-#cmakedefine EVENT__HAVE_TIMERCMP 1
```
[the remark this answers is not in the document]
Comment (azat) [reply event-config.h.cmake:431]: @Coeur let's sync the configure.ac as well

[the remark this answers is not in the document]
Comment (Coeur) [reply event-config.h.cmake:431]: Like that: ccb0c2bcabfa1460bb3c134ddf20ef869e4fc7e9 ?

[the remark this answers is not in the document]
Comment (azat) [reply event-config.h.cmake:431]: Correct, but this is not all, you need to adjust the configure.ac for other changes, i.e. remove HAVE_TAILQFOREACH, and likely something else, I did not look deeply enough for now, please take a look

[the remark this answers is not in the document]
Comment (Coeur) [reply event-config.h.cmake:431]: OK, I removed HAVE_TAILQFOREACH in 8ec6245232c58535bca9122f476ef55d8c6f3804.
Then I manually exhaustively verified all the 150+ definitions of `EVENT__*` and removed the remaining unused ones in f5f6259e817b887be8671b86aa6de75274872e04.

I hope the work was worth it. Please review. :)

[the remark this answers is not in the document]
Comment (azat) [reply event-config.h.cmake:431]: >and likely something else

detection of timercmp is still in configure.ac, please review all the cmake changes, maybe something else should be reflected in autotools builds

>I hope the work was worth it. Please review. :)

Definitely, I like to keep things clean, thank you for your work!

[the remark this answers is not in the document]
Comment (azat) [reply event-config.h.cmake:431]: >detection of timercmp is still in configure.ac

My bad, I checked only last patches. This looks OK.

**Review by azat — APPROVED** [review]

**Pull request merged**